src/Component/src/Doctrine/ORM/Metadata/Resource/Factory/DoctrineORMResourceMetadataCollectionFactory.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Doctrine\ORM\Metadata\Resource\Factory;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+use Sylius\Resource\Doctrine\Common\State\PersistProcessor;
+use Sylius\Resource\Doctrine\Common\State\RemoveProcessor;
+use Sylius\Resource\Metadata\DeleteOperationInterface;
+use Sylius\Resource\Metadata\GridAwareOperationInterface;
+use Sylius\Resource\Metadata\Operation;
+use Sylius\Resource\Metadata\Operations;
+use Sylius\Resource\Metadata\Resource\Factory\ResourceMetadataCollectionFactoryInterface;
+use Sylius\Resource\Metadata\Resource\ResourceMetadataCollection;
+use Sylius\Resource\Metadata\ResourceMetadata;
+
+final class DoctrineORMResourceMetadataCollectionFactory implements ResourceMetadataCollectionFactoryInterface
+{
+    public function __construct(
+        private ManagerRegistry $managerRegistry,
+        private ResourceMetadataCollectionFactoryInterface $decorated,
+    ) {
+    }
+
+    public function create(string $resourceClass): ResourceMetadataCollection
+    {
+        $resourceCollectionMetadata = $this->decorated->create($resourceClass);
+
+        /** @var ResourceMetadata $resource */
+        foreach ($resourceCollectionMetadata->getIterator() as $i => $resource) {
```

Comment:
```suggestion
        foreach ($resourceCollectionMetadata->getIterator() as $i => $resource) {
```
Setting phpstan params (`TKey`, `TValue`) on `ResourceMetadataCollection` would ease type-hinting